monitoring/backup_collector.go

```diff
@@ -0,0 +1,95 @@
+package monitoring
+
+import (
+	"strconv"
+
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/sirupsen/logrus"
+
+	"github.com/longhorn/longhorn-manager/datastore"
+	"github.com/longhorn/longhorn-manager/k8s/pkg/apis/longhorn/v1beta2"
+)
+
+type backupCollector struct {
+	*baseCollector
+
+	stateMetric *metricInfo
+	sizeMetric  *metricInfo
+}
+
+func NewBackupCollector(logger logrus.FieldLogger, nodeID string, ds *datastore.DataStore) prometheus.Collector {
+	return &backupCollector{
+		baseCollector: newBaseCollector(subsystemBackup, logger, nodeID, ds),
+		stateMetric: &metricInfo{
+			Desc: prometheus.NewDesc(
+				prometheus.BuildFQName(longhornName, subsystemBackup, "state"),
+				"State of each backup",
+				[]string{nodeLabel, backupLabel},
+				nil,
+			),
+			Type: prometheus.GaugeValue,
+		},
+		sizeMetric: &metricInfo{
+			Desc: prometheus.NewDesc(
+				prometheus.BuildFQName(longhornName, subsystemBackup, "actual_size_bytes"),
+				"Actual space used by each backup",
+				[]string{nodeLabel, backupLabel},
+				nil,
+			),
+			Type: prometheus.GaugeValue,
+		},
+	}
+}
+
+func (bc *backupCollector) Describe(ch chan<- *prometheus.Desc) {
+	ch <- bc.stateMetric.Desc
+	ch <- bc.sizeMetric.Desc
+}
+
+func (bc *backupCollector) Collect(ch chan<- prometheus.Metric) {
+	backups, err := bc.ds.ListBackups()
```

Comment:
nit: it'd be good to add a new function `ListBackupsRO` inside the datastore function to avoid the DeepCopy all the backups.